bip-0174.mediawiki

```diff
@@ -494,6 +494,39 @@ The currently defined per-input types are defined as follows:
 | 0, 2
 | [[bip-0371.mediawiki|371]]
 |-
+| ECDSA P2C Key Tweak
+| <tt>PSBT_IN_ECDSA_P2C_TWEAK = 0x19</tt>
+| <tt><pubkey></tt>
+| 33 bytes of compact public key serialization specifying to which of keys the ECDSA tweak must be applied (i.e. this MUST be a value of a public key before the tweak is applied)
+| <tt><tweak></tt>
+| The 32 byte value which MUST be added to a private key to produce correct ECDSA signature ("key tweak"). Signers should remove this field after <tt>PSBT_IN_PARTIAL_SIG</tt> is constructed.
+|
+|
+| 0, 2
+| 174
+|-
+| Taproot Per-key P2C Tweak
+| <tt>PSBT_IN_TAP_P2C_TWEAK = 0x1a</tt>
+| <tt><xonlypubkey></tt>
+| 32 bytes of public key specifying to which of keys tweak must be applied (i.e. this MUST be a value of a public key before the tweak is applied)
+| <tt><tweak></tt>
+| The 32 byte value which MUST be added to a private key to produce correct BIP-340 signature ("key tweak"). Signers should remove this field after <tt>PSBT_IN_PARTIAL_SIG</tt> is constructed.
+|
+|
+| 0, 2
+| 174
+|-
+| Taproot Internal P2C Tweak
+| <tt>PSBT_IN_INTERNAL_P2C_TWEAK = 0x1b</tt>
```

Comment:
The difference between this and `PSBT_IN_TAP_P2C_TWEAK` is not clear to me. ISTM this is for the internal key specifically, but the same effect can be achieved by specifying the internal key in `PSBT_IN_TAP_P2C_TWEAK`.

Reply:
Agree, good point